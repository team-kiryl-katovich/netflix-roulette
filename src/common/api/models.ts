export type Result<T> = {
  data?: T;
  error?: Error;
};

export type Offset = {
  offset?: number;
  limit?: number;
};

export type OffsetResult<T> = Result<T> &
  Offset & {
    total?: number;
  };

export type Movie = {
  id: number;
  title: string;
  tagline: string;
  vote_average: number;
  vote_count: number;
  release_date: string;
  poster_path: string;
  overview: string;
  budget: number;
  revenue: number;
  runtime: number;
  genres: string[];
};

export type SortBy = keyof Movie;
export type SortOrder = 'desc' | 'asc';
export type SearchBy = Extract<keyof Movie, 'title' | 'genres'>;

export type MovieFilter = {
  sortBy?: SortBy;
  sortOrder?: SortOrder;
  search?: string;
  searchBy?: SearchBy;
  filter?: string;
};
