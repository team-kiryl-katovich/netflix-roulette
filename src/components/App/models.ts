import { Store } from 'redux';

import { AppState } from '../../store/models';

export type AppProps = {
  store: Store<AppState>;
};
