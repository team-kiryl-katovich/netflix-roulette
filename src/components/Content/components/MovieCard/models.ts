export type MovieCardProps = {
  id: number;
  title: string;
  gengre: string;
  year: number;
  imageUrl: string;
};
