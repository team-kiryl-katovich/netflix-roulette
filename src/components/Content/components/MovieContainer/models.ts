import { WithMoviesProps } from 'src/enhancers/withMovies';

export type MovieContainerComponentProps = WithMoviesProps;
