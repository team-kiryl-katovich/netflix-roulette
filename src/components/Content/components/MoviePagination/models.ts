import { Movie, Offset } from '@common/api';

export type MoviePaginationProps = {
  moviesCount: number;
  offset: Offset;
  pageSize: number;
  loading: boolean;
  children?: React.ReactNode;
  fetch: (offset: Offset) => void;
};

export type MoviePaginationState = {};
