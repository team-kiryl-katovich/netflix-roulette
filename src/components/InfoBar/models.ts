import { WithFilterProps } from '@enhancers/withFilter';
import { WithMoviesProps } from '@enhancers/withMovies';

export type InfoBarComponentProps = WithFilterProps & WithMoviesProps;
