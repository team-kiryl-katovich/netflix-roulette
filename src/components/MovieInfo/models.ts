import { WithMovieProps } from '@enhancers/withMovie';

export type MovieInfoComponentProps = WithMovieProps;
