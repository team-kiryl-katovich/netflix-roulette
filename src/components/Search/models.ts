import { WithMoviesHandlersProps } from '@enhancers/withMovies';
import { WithFilterProps } from '@enhancers/withFilter';

export type SearchComponentProps = WithFilterProps & WithMoviesHandlersProps;
