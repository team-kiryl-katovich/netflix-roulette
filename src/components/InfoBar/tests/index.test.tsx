import * as React from 'react';
import * as enzyme from 'enzyme';

import { InfoBarComponentProps } from '../models';
import { InfoBarComponent } from '..';

describe('[COMPONENTS] [INFO_BAR_COMPONENT]', () => {
  const fakeProps: InfoBarComponentProps = {
    filter: {
      search: 'fakeSearch',
      sortBy: 'release_date',
    },
    moreMovies: jest.fn(),
    moviesData: {
      error: null,
      loading: false,
      movies: [],
      offset: {
        limit: 1,
        offset: 1,
      },
    },
    searchMovies: jest.fn(),
    setFilter: jest.fn(),
  };

  test('should render component', () => {
    const component = enzyme.shallow(<InfoBarComponent {...fakeProps} />);
    expect(component).toMatchSnapshot();
  });

  test('should sort change', () => {
    (fakeProps.setFilter as jest.Mock).mockReset();
    (fakeProps.searchMovies as jest.Mock).mockReset();
    const component = enzyme.mount(<InfoBarComponent {...fakeProps} />);

    const button = component.find('button.netflix-button--secondary');
    button.simulate('click');
    expect(fakeProps.setFilter).toHaveBeenCalledTimes(1);

    component.setProps({
      filter: {
        ...fakeProps.filter,
        sortBy: 'rating',
      },
    });
    expect(fakeProps.searchMovies).toHaveBeenCalledTimes(1);
  });
});
