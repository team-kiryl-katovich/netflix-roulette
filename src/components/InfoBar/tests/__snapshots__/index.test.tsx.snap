// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENTS] [INFO_BAR_COMPONENT] should render component 1`] = `
<div
  className="info-bar"
>
  <div
    className="info-bar__controls"
  >
    <Component
      className="info-label"
      text="0 movies found"
    />
    <Component
      onChange={[Function]}
      options={
        Array [
          Object {
            "clientId": "sort-options-0-0",
            "item": Object {
              "title": "Release date",
              "value": "release_date",
            },
          },
          Object {
            "clientId": "sort-options-0-1",
            "item": Object {
              "title": "Rating",
              "value": "vote_average",
            },
          },
        ]
      }
      title="Sort by"
      value="release_date"
    />
  </div>
</div>
`;
